Dockerfile

```diff
@@ -44,54 +71,59 @@ ENV GDAL_DATA=/opt/conda/lib/python${PYTHON_VERSION}/site-packages/rasterio/gdal
+ENV PYTHONPATH=/opt/src:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_pipeline/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_aws_s3/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_aws_batch/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_gdal_vsi/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_core/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_pytorch_learner/:$PYTHONPATH
+ENV PYTHONPATH=/opt/src/rastervision_pytorch_backend/:$PYTHONPATH
 
-COPY ./rastervision_core/requirements.txt /opt/src/requirements.txt
-RUN pip install $(grep -ivE "^\s*$|^#|rastervision_*" requirements.txt)
+WORKDIR /opt/src/
 
-COPY ./rastervision_pytorch_learner/requirements.txt /opt/src/requirements.txt
-RUN pip install $(grep -ivE "^\s*$|^#|rastervision_*" requirements.txt)
+#------------------------------------------------------------------------
 
-COPY ./rastervision_gdal_vsi/requirements.txt /opt/src/requirements.txt
-RUN pip install $(grep -ivE "^\s*$|^#|rastervision_*" requirements.txt)
+COPY ./rastervision_gdal_vsi/requirements.txt /opt/src/gdal-requirements.txt
+COPY ./rastervision_aws_s3/requirements.txt /opt/src/s3-requirements.txt
+COPY ./rastervision_aws_batch/requirements.txt /opt/src/batch-requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip cat gdal-requirements.txt s3-requirements.txt batch-requirements.txt | sort | uniq > all-requirements.txt && \
+    pip install $(grep -ivE "^\s*$|^#|rastervision_*" all-requirements.txt) && \
+    rm all-requirements.txt
 
-# Commented out because there are no non-RV deps and it will fail if uncommented.
-# COPY ./rastervision_pytorch_backend/requirements.txt /opt/src/requirements.txt
-# RUN pip install $(grep -ivE "^\s*$|^#|rastervision_*" requirements.txt)
+COPY ./rastervision_pipeline/requirements.txt /opt/src/pipeline-requirements.txt
+COPY ./rastervision_core/requirements.txt /opt/src/core-requirements.txt
+COPY ./rastervision_pytorch_learner/requirements.txt /opt/src/pytorch-requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip cat pipeline-requirements.txt core-requirements.txt pytorch-requirements.txt | sort | uniq > all-requirements.txt && \
+    pip install $(grep -ivE "^\s*$|^#|rastervision_*" all-requirements.txt) && \
+    rm all-requirements.txt
 
-#########################
-# Docs
-#########################
-# Install docs/requirements.txt
-COPY ./docs/requirements.txt /opt/src/docs/requirements.txt
-RUN pip install -r docs/requirements.txt
+COPY ./requirements-dev.txt /opt/src/requirements-dev.txt
+RUN --mount=type=cache,target=/root/.cache/pip pip install -r requirements-dev.txt
 
-# Install pandoc, needed for rendering notebooks
-# Get latest release link from here: https://github.com/jgm/pandoc/releases
-ARG TARGETARCH
-RUN wget https://github.com/jgm/pandoc/releases/download/2.19.2/pandoc-2.19.2-1-${TARGETARCH}.deb && \
+# pandoc
+COPY ./docs/requirements.txt /opt/src/docs/pandoc-requirements.txt
```

Comment:
These are not pandoc requirements, but rather general requirements for building RV docs.

Reply:
:heavy_check_mark: